libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2475,36 +2475,45 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 	});
 }
 
-string YulUtilFunctions::bytesConcatFunction(vector<Type const*> const& _argumentTypes)
+string YulUtilFunctions::bytesOrStringConcatFunction(
+	vector<Type const*> const& _argumentTypes,
+	FunctionType::Kind functionTypeKind
+)
 {
-	string functionName = "bytes_concat";
+	solAssert(functionTypeKind == FunctionType::Kind::BytesConcat || functionTypeKind == FunctionType::Kind::StringConcat);
+	std::string functionName = (functionTypeKind == FunctionType::Kind::StringConcat) ? "string_concat" : "bytes_concat";
 	size_t totalParams = 0;
 	vector<Type const*> targetTypes;
+
 	for (Type const* argumentType: _argumentTypes)
 	{
-		solAssert(
-			argumentType->isImplicitlyConvertibleTo(*TypeProvider::bytesMemory()) ||
-			argumentType->isImplicitlyConvertibleTo(*TypeProvider::fixedBytes(32)),
-			""
-		);
+		if (functionTypeKind == FunctionType::Kind::StringConcat)
+			solAssert(argumentType->isImplicitlyConvertibleTo(*TypeProvider::stringMemory()));
+		else if (functionTypeKind == FunctionType::Kind::BytesConcat)
+			solAssert(
+				argumentType->isImplicitlyConvertibleTo(*TypeProvider::bytesMemory()) ||
+				argumentType->isImplicitlyConvertibleTo(*TypeProvider::fixedBytes(32))
+			);
+
 		if (argumentType->category() == Type::Category::FixedBytes)
-			targetTypes.emplace_back(argumentType);
+				targetTypes.emplace_back(argumentType);
```

Comment:
```suggestion
			targetTypes.emplace_back(argumentType);
```
That's too much indentation, isn't it?